### PR TITLE
adjust ca handling for helm repositories

### PR DESCRIPTION
**Pull request by fabianburth**
**What this PR does / why we need it**:
To implement a proper handling of custom certificate authorities for interaction with helm repositories.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release Notes**:

```feature user

```

### CODE REVIEW

File: pkg/helm/downloader.go [hunk 1]
```diff
@@ -139,8 +138,8 @@ func (d *chartDownloader) complete(ctx oci.ContextProvider, ref, repourl string)
 		return err
 	}
 
-	if len(d.cacert) != 0 {
-		err = d.writeFile("cacert", config, &entry.CAFile, d.cacert, "CA file")
+	if len(creds[identity.ATTR_CERTIFICATE_AUTHORITY]) != 0 {
+		err = d.writeFile("ca", config, &entry.CAFile, []byte(creds[identity.ATTR_CERTIFICATE_AUTHORITY]), "CA file")
```
Comment (mandelsoft) [inline pkg/helm/downloader.go:142]: may be we should keep the filename `cacert`

**Review by mandelsoft — CHANGES_REQUESTED** [review]

File: pkg/helm/chartaccess.go [hunk 2]
```diff
@@ -50,17 +50,23 @@ type chartAccess struct {
 
 var _ ChartAccess = (*chartAccess)(nil)
 
+func newChartAccess(path string, fss ...vfs.FileSystem) (*chartAccess, error) {
```
Comment (Skarlso) [inline pkg/helm/chartaccess.go:53]: What is the purpose of this extracted function? It isn't called anywhere else other than the temp access below which already does the thing this does. :) Is there a plan to call this from something else?

Comment (Skarlso) [reply pkg/helm/chartaccess.go]: Neat. Thanks.

Comment (fabianburth) [reply pkg/helm/chartaccess.go]: Initially, we didn't realise we had access to the file system of the chart access from the downloader. Therefore, we thought we needed an option to pass in an arbitrary file system. This is pretty much a remnant of that original thought. As it is currently not necessary, I removed it for now.

**Review by mandelsoft — APPROVED** [review]

**Pull request merged**